### PR TITLE
Adding new query param and tests

### PR DESCRIPTION
**Pull request by RedBrogdon**
* Updates `GistLoader` to take a bool indicating whether to use the master server when loading from the API Doc.
* Adds a new query parameter to embed.dart called `sample_channel`
  - If it's set to "master", the embed will tell the `GistLoader` to use the master server.
  - If it's anything else (or missing), the stable server is used.
* Updates tests based on this new, larger reality in which we find ourselves.

### CODE REVIEW

**Comment by RedBrogdon** [comment]
CC @gspencergoog.

File: lib/embed.dart [hunk 1]
```diff
@@ -573,7 +581,8 @@ class Embed {
       if (gistId.isNotEmpty) {
         gist = await loader.loadGist(gistId);
       } else if (sampleId.isNotEmpty) {
-        gist = await loader.loadGistFromAPIDocs(sampleId);
+        final useMaster = sampleChannel?.toLowerCase() == 'master';
+        gist = await loader.loadGistFromAPIDocs(sampleId, useMaster);
```
Comment (gspencergoog) [inline lib/embed.dart]: Why not pass the string 'master' if `sampleChannel` is 'master', and 'stable' if it isn't.  Obviously we don't want to just pass through the query param, but if the argument is a string, `loadGistFromAPIDocs` will be more future proof if we do add more channels later.

Comment (RedBrogdon) [reply lib/embed.dart]: I thought about that, but figured that if we're unlikely to need more than the two choices, passing a boolean was cleaner than passing a string.

Is it more idiomatic to Dart to use a string there?

Comment (gspencergoog) [reply lib/embed.dart]: Well, most idiomatic would probably be an enum, which would also help make sure no cases are missed in the future.

Comment (RedBrogdon) [reply lib/embed.dart]: Discussed offline with @johnpryan. We're going to need an enum for SDK channel eventually, so I've gone that route.

**Pull request merged**